### ISSUE TITLE
<Focused FX> and also filter by name 

### ISSUE THREAD

**Issue by TommyC81** [helgoboss/realearn#610]
I can't seem to find an option to fit my use case: I want to prepare one generic preset that captures endless rotary mappings for the macros in my most commonly used software synths.

However, using "Focused FX" as target makes it hard to filter between which synth it is i.e. the parameter mappings differ and causes issues.

Using Track "Selected" allows me to filter the FX "By name" and apply the correct mappings. However, this requires me to select the correct track, not only focus the desired FX - this can be very confusing when having multiple windows open.

An option to use "Focused FX" and also filter "By name" would achieve the desired result i.e. I can simply activate the FX window I want to use and start using my controller. However, this does not seem to be possible currently.

For example; I have "Pigments" (synth) on a track, when I select that track I can then provide input to the "*Pigments*" FX on that track. But when I have several tracks with Pigments and other synths on them (Synthmaster and Vital, for instance), and the corresponding windows open, there is no easy way to change input as I must change focused track, not just select my window. Ideally, I could just set "Focused FX" and then apply correct midi settings/parameters based on the FX name (the parameter names/IDs differ significantly between the different synths).

Maybe this is possible, but I can't find (an easy) way to achieve this in UI nor documentation.

**Comment by helgoboss** [comment]
Mmh, I don't really follow to be honest. Maybe you look for "Auto-load preset" feature? There's a tutorial for this in the tutorials section of the user guide.

If not, could you rephrase?

**Comment by TommyC81** [comment]
Apologies, I've completely overlooked the auto-load feature. It's likely what I'm looking for.

I will close this ticket, and re-open a new one with a better description (if a new ticket is even required). I'm away from my computer at the moment, so I'm typing all from phone and memory unfortunately.

Thank you!

**Comment by helgoboss** [comment]
No problem. BTW, "Auto-load preset" is getting an overhaul (#188, #228), which will be available in the next pre-release 2.13.0-pre.9. Then it will be possible to load a different preset not just based on which FX is currently focused but also based on what is the current "Instance FX" (for that ReaLearn instance).

The "Instance FX" can be anything, here are just a few examples:

- The second FX on the currently selected track.
- The first FX named "Vital" on the track below the currently selected track.
- A previously pinned FX (there are new targets for pinning a certain FX).
- ...

**Comment by TommyC81** [comment]
The auto-load feature seems to work great! I'm truly impressed about the
flexibility and capabilities of ReaLearn. Only yesterday found your video
about ReaLearn 2 - great and interesting viewing!
